autogpt_platform/backend/backend/util/feature_flag.py

```diff
@@ -252,6 +285,75 @@ async def async_wrapper(*args: P.args, **kwargs: P.kwargs) -> T:
     return decorator
 
 
+def create_feature_flag_dependency(
+    flag_key: Flag,
+    default: bool = False,
+) -> Callable[[str | None], Awaitable[None]]:
+    """
+    Create a FastAPI dependency that checks a feature flag.
+
+    This dependency automatically extracts the user_id from the JWT token
+    (if present) for proper LaunchDarkly user targeting, while still
+    supporting anonymous access.
+
+    Args:
+        flag_key: The Flag enum value to check
+        default: Default value if flag evaluation fails
+
+    Returns:
+        An async dependency function that raises HTTPException if flag is disabled
+
+    Example:
+        router = APIRouter(
+            dependencies=[Depends(create_feature_flag_dependency(Flag.CHAT))]
+        )
+    """
+
+    async def check_feature_flag(
```

Comment:
shouldn't this a private helper like _get_optional_user_id_from_auth ?

Reply:
Using a function that does the same thing from autogpt_libs module